### ISSUE TITLE
Allow querying by annotations that don't have a given phenotype condition

### ISSUE THREAD

**Issue by kimrutherford** [pombase/website#1463]
See #56

Is it useful to be able to query for eg. NOT high temperature?

**Comment by ValWood** [comment]
Note this will be useful because for example, if you want all standard growth conditions or standard temperature, we would not always explicitly capture this. The best way to get the most accurate results would be by an  exclusion query.

**Comment by kimrutherford** [comment]
The server-side code is done to support this.  We just need to decide what the interface should look like.

> The best way to get the most accurate results would be by an exclusion query.

I don't understand this bit.

**Comment by ValWood** [comment]
Probably what you have done.

When querying on phenotypes users might want to exclude all "high temperature" and "low temperature" for example, than to select all "standard temperature" because we often would not add the conditions explicitly if they are standard.

**Comment by kimrutherford** [comment]
> We just need to decide what the interface should look like.

Would this be OK with improved wording?:
![exclude-conditions-0](https://user-images.githubusercontent.com/90474/78131449-96505a00-7477-11ea-898e-8225cd7f4227.png)
![exclude-conditions-1](https://user-images.githubusercontent.com/90474/78131469-9b150e00-7477-11ea-99fa-f2ca40263de2.png)

**Comment by ValWood** [comment]
Well having the  text box is there by default would make the feature clearer, and probe boy encourage it's use.....
Anyone?

**Comment by kimrutherford** [comment]
As discussed, I've released the changes so far into the dev version of the site.  The main site is unchanged.  The change in dev adds a "Exclude condition" option for phenotypes.

We should think about how to extend this to multiple conditions so users can do things like exclude "high temperature" and "low temperature" simultaneously.

**Comment by ValWood** [comment]
I tested this and it seems to work as I expeced.


An example that might be useful  to explain that this is annotation beased and not gene based


40 | abnormal mitotic spindle elongation (FYPO:0000131) [single allele genotypes]

38 | abnormal mitotic spindle elongation (FYPO:0000131) [single allele genotypes - excluded conditions: low temperature (PECO:0000006)]

this query will exclude dis3 and scn1 which have  low temperature ONLY

but will include  dis1 which has

FYPO:0002688 | mitotic spindle collapse without elongation during prophase | dis1Δ | Microscopy | low temperature | Hsu KS et al. (2011)

and

FYPO:0007071 | increased rate of mitotic spindle elongation during anaphase A | dis1Δ | Microscopy |   | Pinder C et al. (2019)

(no temperature recorded)


















--

**Comment by ValWood** [comment]
To make this really clear maybe the tex could say

Constrain annotations with condition:
Exclude annotations with condition: 

A bit longer but probably worth it?

**Comment by ValWood** [comment]
AND
Where we say
Include genes from:

should this say
Include genes with annotations from:

**Comment by ValWood** [comment]
Looks pretty good to me. I had quite a good play around and the behaviur is as I expected ... Will wait for second opinion from @mah11 , and on the text tweaks.

**Comment by mah11** [comment]
> Constrain annotations with condition:
> Exclude annotations with condition:
> 
> A bit longer but probably worth it?

Yes, although perhaps now that we can specify conditions to include OR exclude we could avoid saying "constrain":

Include annotations with condition:
Exclude annotations with condition:

> Include genes from:
> 
> should this say
> Include genes with annotations from:

I don't think we'd gain nearly as much by changing this bit.

**Comment by ValWood** [comment]
Action: implement text changes and release
Action: Allow multiple conditions to be included or excluded, as per 
https://github.com/pombase/website/issues/1462
Action: Update documentation

**Comment by kimrutherford** [comment]
I think this is all implemented now.

**Comment by ValWood** [comment]
note to self Check if it is clear how this works and if it 's documented.

**Comment by ValWood** [comment]
Will reopen

@PCarme  can you check if the "NOT" queries work as you would expect .

We might need documentation to explain anything that might be unclear.
Also it's still an open question whether to use "contain" or include/exclude (which is probably simpler)

**Comment by ValWood** [comment]
Also because this query is gene-based/ not allele based, if you make a query

if a gene is annotated to
inviable  standard temperature
inviable  high temperature
does it remove the gene, even though it is inviabale at  standard temperature (I think it would).
In this case we need to explain this, because. it may not be what would be expected.

**Comment by ValWood** [comment]
Actually I think the docs are quite comprehensive
https://www.pombase.org/documentation/advanced-search#constraining-by-conditions

The “Exclude condition” option excludes from the results any annotations to the phenotype term of interest that include the specified condition (note: annotations to the same gene with a different condition, or no condition specified, will retained in the gene result list). This means, for example, if you exclude “high temperature” for a specific phenotype, the gene will remain in your output list if it has another annotation to the same term without a recorded condition, or “standard temperature”. To find a condition, start typing, then choose from the autocomplete options.

Warning: Condition constraint and exclusion should be used with caution, and will be more successful for non-standard conditions like “high temperature” and “low temperature” “low glucose” because standard conditions and media are not always explicitly recorded.

It it is cleverer than I remembered"(note: annotations to the same gene with a different condition, or no condition specified, will retained in the gene result list)."

@PCarme  if you can check that the behaviusr matches the docs

@kimrutherford  can you split the docs (add a new heading "excluded conditions"  so that the links go directly to the sections describing these two query filters (at the moment, it is lumped with constrained.

Then we only need to decide  "constrained" or "included" in the  query and docs?

**Comment by kimrutherford** [comment]
> can you split the docs (add a new heading "excluded conditions" so that the links go directly to the sections describing these two query filters (at the moment, it is lumped with constrained.

Good plan.  I've done that and the change should be visible in the morning.

**Comment by PCarme** [comment]
I tried different queries and combinations, it behaves as described in the documentation. 

Maybe we should state more clearly that the "exclude condition" queries will still return genes that have annotations to the specified phenotype and condition if they also have annotations to this phenotype without the condition (in other words, the "exclude condition" doesn't work as a NOT query). It might just be me, but I got confused when I started testing it...

**Comment by kimrutherford** [comment]
> It might just be me, but I got confused when I started testing it..

That's definitely the most confusing part and it's quite difficult to explain clearly.
